packages/treat/src/validateSelector.test.js

```diff
@@ -0,0 +1,56 @@
+const { validateSelector } = require('../lib/commonjs/validateSelector');
+
+describe('validateSelector', () => {
+  describe('selectors that target "&"', () => {
+    const validSelectors = [
+      '&',
+      '& &',
+      '.foo &',
+      '[foo] &',
+      '& .foo &',
+      '.foo ~ .bar &',
+      '.foo + .bar &',
+      '.foo > .bar &',
+      '&:hover',
+      '&:not(:disabled)',
+      '&.foo',
+      '&.foo.bar',
+      '.foo&',
+      '.foo.bar&',
+      '.foo.bar.baz&',
+      '&.foo:hover',
+      '&.foo.bar:hover',
+      '&.foo.bar:not(.baz)',
+      '.foo &:not(:disabled)',
+      '.foo ~ .bar > &:not(:disabled)',
```

Comment:
Worth having a test for the `*` selector?

Reply:
Done. Also added some test cases with commas, since I realised I missed those too.